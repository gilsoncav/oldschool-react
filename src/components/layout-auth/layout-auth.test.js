import LayoutAuth from './layout-auth';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

const mockStore = configureStore();

describe('LayoutAuth component', () => {
  it('renders correctly', () => {
    const store = mockStore({
      router: { location: { pathname: '/dashboard' } },
      auth: { isLoadingLogout: false },
      sider: { isCollapsed: true },
    });

    const snapshot = snapRender(
      <MemoryRouter>
        <Provider store={store}>
          <LayoutAuth>TESTE</LayoutAuth>
        </Provider>
      </MemoryRouter>,
    );

    expect(snapshot).toMatchSnapshot();
  });
});
