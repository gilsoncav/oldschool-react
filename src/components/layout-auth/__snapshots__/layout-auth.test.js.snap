// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutAuth component renders correctly 1`] = `
<section
  className="ant-layout"
  id="layout-auth"
>
  <header
    className="ant-layout-header"
    style={
      Object {
        "paddingLeft": "5px",
        "paddingRight": "5px",
        "textAlign": "center",
      }
    }
  >
    <div
      className="logo"
      style={
        Object {
          "float": "left",
        }
      }
    >
      <button
        className="ant-btn btn-secondary ant-btn-lg ant-btn-icon-only"
        id="toggle-sider"
        onClick={[Function]}
        type="button"
      >
        <i
          aria-label="icon: menu-unfold"
          className="anticon anticon-menu-unfold"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="menu-unfold"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 0 0 0-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0 0 14.4 7z"
            />
          </svg>
        </i>
      </button>
      <a
        href="/dashboard"
        onClick={[Function]}
      >
        <img
          alt="Logo"
          src="logo.png"
          style={
            Object {
              "marginLeft": "15px",
              "marginRight": "30px",
              "width": "120px",
            }
          }
        />
      </a>
    </div>
    <div
      className="logo"
      style={
        Object {
          "float": "right",
        }
      }
    >
      <button
        className="ant-btn btn-secondary ant-btn-lg ant-btn-icon-only"
        id="logout-btn"
        onClick={[Function]}
        type="button"
      >
        <i
          aria-label="icon: logout"
          className="anticon anticon-logout"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="logout"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 0 1-112.7 75.9A352.8 352.8 0 0 1 512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 0 1-112.7-75.9 353.28 353.28 0 0 1-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 0 0 0-12.6z"
            />
          </svg>
        </i>
      </button>
    </div>
    <div
      style={
        Object {
          "float": "right",
          "marginRight": "35px",
        }
      }
    >
      <div
        id="audio-toggle"
      >
        <h4
          className="text-center text-primary"
          style={
            Object {
              "display": "inline",
              "marginRight": "10px",
            }
          }
        >
          Playing: 
          top-gear.mp3
        </h4>
        <button
          className="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg"
          onClick={[Function]}
          style={
            Object {
              "marginRight": "10px",
            }
          }
          type="button"
        >
          <i
            aria-label="icon: backward"
            className="anticon anticon-backward"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="backward"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M485.6 249.9L198.2 498c-8.3 7.1-8.3 20.8 0 27.9l287.4 248.2c10.7 9.2 26.4.9 26.4-14V263.8c0-14.8-15.7-23.2-26.4-13.9zm320 0L518.2 498a18.6 18.6 0 0 0-6.2 14c0 5.2 2.1 10.4 6.2 14l287.4 248.2c10.7 9.2 26.4.9 26.4-14V263.8c0-14.8-15.7-23.2-26.4-13.9z"
              />
            </svg>
          </i>
        </button>
        <button
          className="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg ant-btn-icon-only"
          id="audio-btn"
          onClick={[Function]}
          type="button"
        >
          <i
            aria-label="icon: caret-right"
            className="anticon anticon-caret-right"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="caret-right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
              />
            </svg>
          </i>
        </button>
        <button
          className="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "10px",
            }
          }
          type="button"
        >
          <i
            aria-label="icon: forward"
            className="anticon anticon-forward"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="forward"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M825.8 498L538.4 249.9c-10.7-9.2-26.4-.9-26.4 14v496.3c0 14.9 15.7 23.2 26.4 14L825.8 526c8.3-7.2 8.3-20.8 0-28zm-320 0L218.4 249.9c-10.7-9.2-26.4-.9-26.4 14v496.3c0 14.9 15.7 23.2 26.4 14L505.8 526c4.1-3.6 6.2-8.8 6.2-14 0-5.2-2.1-10.4-6.2-14z"
              />
            </svg>
          </i>
        </button>
      </div>
    </div>
  </header>
  <section
    className="ant-layout ant-layout-has-sider"
  >
    <aside
      className="ant-layout-sider ant-layout-sider-dark ant-layout-sider-collapsed ant-layout-sider-zero-width"
      style={
        Object {
          "flex": "0 0 0px",
          "maxWidth": "0px",
          "minWidth": "0px",
          "width": "0px",
        }
      }
    >
      <div
        className="ant-layout-sider-children"
      />
    </aside>
    <main
      className="ant-layout-content"
    >
      TESTE
    </main>
  </section>
  <footer
    className="ant-layout-footer"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <span>
      2019 - Kevin Aguiar - 
      <a
        href="https://github.com/kevinfaguiar"
      >
        Github
      </a>
    </span>
  </footer>
</section>
`;
